src/core/prototypes/plot_segments/region_2d/utils.ts

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { PolarAttributes, PolarProperties } from "./polar";
+
+export enum SIDE {
+  HORIZONTAL,
+  VERTICAL,
+  NONE,
+}
+
+interface CenterType {
+  cx: number;
+  cy: number;
+  side: SIDE;
+  isHalf: boolean;
+  ratio: number;
+  isRightHalf: boolean;
+}
+
+// eslint-disable-next-line max-lines-per-function
+export function getCenterByAngle(
+  properties: PolarProperties,
+  attrs: PolarAttributes
+): CenterType {
+  const isAutoMargin = properties.autoAlignment ?? false;
+  const { angle1, angle2, x1, y1, x2, y2 } = attrs;
+  let cx;
+  let cy;
+  let radialRatio = 1;
+  let isHalf = false;
+  let side = SIDE.NONE;
+  let isRightHalf = false;
+  if (isAutoMargin) {
+    //pos case
+
+    const angleDelta = Math.abs(angle1 - angle2);
+    const startAngle = angle1 % 360;
+
+    if (startAngle >= 0 && startAngle < 90) {
+      //startAngle - 1 quadrant
+      if (angleDelta <= 90 - startAngle) {
+        //endAngle - 1 quadrant => move left bottom corner
+        cx = x1;
+        cy = y1;
+        radialRatio = 2;
+      } else if (angleDelta <= 180 - startAngle) {
+        //endAngle - 4 quadrant => move left
+        cx = x1;
+        cy = (y2 + y1) / 2;
+        // radialRatio = 2;
+        isHalf = true;
+        side = SIDE.VERTICAL;
+      } else if (angleDelta + startAngle > 180) {
+        //endAngle - 3 and 4 quadrants
+        cx = (x2 + x1) / 2;
+        cy = (y2 + y1) / 2;
```

Comment:
Should `isHalf` flag be `true` for this condition?

Reply:
For this case `isHalf` flag should be `false`. The chart takes more than 180 degrees.